common/defs.go

```diff
@@ -82,6 +89,76 @@ type PortUpdateMessage struct {
 	Port uint64 `json:"port"`
 }
 
+type Endpoint struct {
+	CidrStr string `json:"cidr,omitempty"`
+	// TODO this can be collapsed into Cidr but needs
+	// work on JSON marshaller/unmarshaller to do that.
+	Cidr      net.IPNet `json:"-"`
+	TenantID  uint64    `json:"tenant_id,omitempty"`
+	TenantExternalID string  `json:"tenant_external_id,omitempty"`
+	SegmentID uint64    `json:"segment_id,omitempty"`
+	SegmentExternalID string `json:"segment_external_id,omitempty"`
+}
+
+const (
+	PolicyDirectionIngress = "ingress"
+	PolicyDirectionEgress  = "egress"
+)
+
+type PortRange [2]uint
+
+type Rule struct {
+	Protocol   string
+	Ports      []uint
+	PortRanges []PortRange
+	IcmpType   uint
+	IcmpCode   uint
+	Stateful   bool
+}
+
+type Rules []Rule
+
+// Metadata attached to entities for various external environments like openstack/kybernetes
+type Tag struct {
+	Key   string
+	Value string
+}
+
+type Policy struct {
+	Direction   string     `json:"direction,omitempty"`
+	Description string     `json:"description,omitempty"`
+	Name        string     `json:"name,omitempty"`
```

Comment:
I think shortening chains is not really an option since limits in policy names are going to be there anyway. 3 letter or 6 or 9 - it's a limit we're imposing on user input and they are going to go creative at some point.
Agree with @debedb, we need a unique id for a policy, hex encoded sequence works for me.